### PR TITLE
[BAAI] vllm generate inference benchmark

### PR DESCRIPTION
**Pull request by daliwang777**

### CODE REVIEW

**Review by tianxiao-baai — APPROVED** [review]

File: inference_generate/vllm_nvs/nvstandard/gettime.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            #     inputs.input_ids,
+            #     max_new_tokens=1,  # 控制生成的额外令牌数
+            #     num_return_sequences=1,  # 返回序列数量
+            #     no_repeat_ngram_size=2,  # 避免重复的n-gram
+            #     early_stopping=True , # 如果达到良好的结束点则停止生成
+            #     attention_mask=inputs.attention_mask,
+            #     num_beams=5,
+            #     top_k=50
+            # )
+            outputs=model.generate(inputs, **params1)
```
Comment (KerwinKai) [inline inference_generate/vllm_nvs/nvstandard/gettime.py:60]: 这里是指vllm载入model时，同时绑定tokenizer吗，因为看这里接受的输入时str

File: inference_generate/vllm_nvs/nvstandard/gettime.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    for i in tqdm(range(10),desc="inference"):
+        with torch.no_grad():
+            timest=time.time()
+            input_text=tasks[i]
+            logger.info(f"--------------------------------------------------------------------")
+            logger.info(f"input lenth is: {len(input_text)}")
+            #print("input lenth is: ", len(input_text))
+            # inputs=tokenizer([input_text+"Please generate a summary about it:"],return_tensors="pt")
+            inputs =input_text+"Please generate a summary about it:"
+            inputs=inputs.to(device)
```
Comment (KerwinKai) [inline inference_generate/vllm_nvs/nvstandard/gettime.py:49]: 这里是str2cuda么

Comment (114homo514) [reply inference_generate/vllm_nvs/nvstandard/gettime.py:49]: 是的 这个框架是针对hf接口下的

File: generate_inference/host.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+DATAPATH: "/home/daliwang/FlagPerf/newinference/inference_generate/data/Xsum/dev.csv"
```
Comment (KerwinKai) [inline generate_inference/host.yaml:1]: 隐藏路径信息，建议改为/path/to/dev.csv

Comment (daliwang777) [reply generate_inference/host.yaml:1]: okok

Comment (daliwang777) [reply generate_inference/host.yaml]: 这里修改完毕了

**Comment by shh2000** [comment]
1. you should remove all __pycache__ files
2. like training and inference in FlagPerf, you should add 2 READMEs: 1 at generate_inference/ to introduce datasets/ckpts and their download methods, 1 at TTFT/<vendor>/<engine>/ to display results
3. you should use yapf -i --style "pep8" --recursive generate_inference to reformat, or using similar tools in vscode/pycharm
4. you should move config_analysize(which should be analyze) and result_show.py into utils/ only left main.py and yamls
5. you'd better place GPUconfig into <TASK>/<vendor>/<engine>/, and read the information automatically by main.py, which is no need to determined at host.yaml
6. you'd better replace strings like /home/daliwang into /root or /home
7. you should remove useless or confused configname like NUMS, PARAMETERS, using nproc_per_node to replace GPUNUMS
8. you should remove all redundent codes or comments
9. you should add related documents at FlagPerf/README.md and FlagPerf/docs/generate, especially tell vendors how to add their engine or support existed engines on their hardwares
10. you can rename generate_inference to generate

**Pull request closed without merging**